src/Rx/Weno/TransmitProperties.php

```diff
@@ -0,0 +1,213 @@
+                          "CountryCode" => "US",
+                         "PrimaryPhone" => $this->patient['phone_cell'],
+                          "SupportsSMS" => "Y",
+                         "PatientEmail" => $this->patient['email'],
+                        "PatientHeight" => $this->vitals['height'],
+                        "PatientWeight" => $this->vitals['weight'],
+          "HeightWeightObservationDate" => $heighDate[0],
+        "ResponsiblePartySameAsPatient" => 'Y',
+                      "PatientLocation" => "Home",
+                 "PrimaryPharmacyNCPCP" => $this->ncpdp,
+             "AlternativePharmacyNCPCP" => $this->ncpdp
+        ];
+
+        return json_encode($wenObj);
+    }
+
+    /**
+     * @return mixed
+     */
+    public function getProviderEmail()
+    {
+        $provider_info = sqlQuery("select email from users where username=? ", [$_SESSION["authUser"]]);
+        if (empty($provider_info['email'])) {
+            echo xlt('Provider email address is missing');
+            exit;
+        } else {
+            return $provider_info;
+        }
+    }
+
+    /**
+     * @return mixed
+     */
+    public function getFacilityInfo()
+    {
+        $locid = sqlQuery("select name, street, city, state, postal_code, phone, fax, weno_id from facility where id = ?", [isset($_SESSION['facilityId'])]);
+
+        if (empty($locid['weno_id'])) {
+            //if not in an encounter then get the first facility location id as default
+            $default_facility = sqlQuery("select name, street, city, state, postal_code, phone, fax, weno_id from facility order by id asc limit 1");
+
+            if (empty($default_facility)) {
+                echo xlt('Facility ID is missing');
+                exit;
+            } else {
+                return $default_facility;
+            }
+        }
+        return $locid;
+    }
+
+    /**
+     * @return mixed
+     */
+    private function getPatientInfo()
+    {
+        //get patient data
+        $patient = sqlQuery("select title, fname, lname, mname, street, state, city, email, phone_cell, postal_code, dob, sex, pid from patient_data where pid = ?", [isset($_SESSION['pid'])]);
+        if (empty($patient['fname']) || empty($patient['lname']) || empty($patient['dob']) || empty($patient['sex']) || empty($patient['postal_code']) || empty($patient['street']) || empty($patient['email'])) {
+            echo xlt('Patient data is incomplete phone or ')
+                . ", "
+                . xlt('first last name') . ", "
+                . xlt('gender email') . ", "
+                . xlt('zip code') . ", "
+                . xlt('date of birth or address');
+            exit;
+        }
+        return $patient;
+    }
+
+    /**
+     * @return string
+     * New Rx
+     */
+    public function cipherpayload()
+    {
+        $cipher = "aes-256-cbc"; // AES 256 CBC cipher
+        $cryptoGen = new Crypto\CryptoGen();
```

Comment:
since using this a couple times in this class, would initiate this call in the class constructor and then make the calls from that.:
```
$enc_key = $this->cryptoGen->decryptStandard($GLOBALS['weno_encryption_key']);
```
(it's much more efficient to do this when calling it multiple times)

Reply:
done